src/main/java/com/microsoft/azure/vmagent/AzureVMCloudPoolRetentionStrategy.java

```diff
@@ -19,22 +23,25 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-public class AzureVMCloudPoolRetentionStrategy extends AzureVMCloudBaseRetentionStrategy {
+public class AzureVMCloudPoolRetentionStrategy extends AzureVMCloudBaseRetentionStrategy implements ExecutorListener {
     private static final long serialVersionUID = 1577788691L;
 
     private final long retentionMillis;
 
     private final int poolSize;
 
+    private final boolean singleUseAgents;
+
     private static final long IDLE_LIMIT_MILLIS = TimeUnit.MINUTES.toMillis(1);
 
     private static final Logger LOGGER = Logger.getLogger(AzureVMManagementServiceDelegate.class.getName());
 
     @DataBoundConstructor
-    public AzureVMCloudPoolRetentionStrategy(int retentionInHours, int poolSize) {
+    public AzureVMCloudPoolRetentionStrategy(int retentionInHours, int poolSize, boolean singleUseAgents) {
```

Comment:
can you move this to a setter annotated with `@DataBoundSetter` please.

otherwise this is a breaking change for JCasC users as constructor arguments are mandatory.

rest looks good

Reply:
alright, this was new to me so had to make some assumptions on the exact implementations, but now this seems to be correct as far as I can gather.

Reply:
you shouldn't add a deprecated constructor, just remove it as it was never used, (if this was in a released version and there was existing consumers then yes deprecating may be needed)

Reply:
this is still outstanding, mandatory arguments go in the constructor, this should have a setter method added annotated with `DataBoundSetter` otherwise its a breaking change for JCasC users